src/Microsoft.Identity.Client/Mats/Internal/ActionComparer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Identity.Client.Mats.Internal
+{
+    internal static class ActionComparer
+    {
+        public static bool IsEquivalentClass(ActionPropertyBag action1, ActionPropertyBag action2)

```

Comment:
Please add a comment to describe this criteria, reading the code is not obvious.